src/deluge/model/clip/audio_clip.h

```diff
@@ -106,6 +106,18 @@ class AudioClip final : public Clip {
 	bool doingLateStart;
 	bool maySetupCache;
 
+	//START ~ new Automation Clip View Variables
+	bool onAutomationAudioClipView; //new to save the view that you are currently in
+	                                //(e.g. if you leave clip and want to come back where you left off)
+
+	int32_t lastSelectedParamID;       //last selected Parameter to be edited in Automation Instrument Clip View
+	Param::Kind lastSelectedParamKind; //0 = patched, 1 = unpatched, 2 = global effectable, 3 = none
+	int32_t lastSelectedParamShortcutX;
+	int32_t lastSelectedParamShortcutY;
+	int32_t lastSelectedParamArrayPosition;
```

Comment:
These could all move to `Clip` rather than being duplicated in `AudioClip` and `InstrumentClip`, since those are the only 2 subclasses of `Clip`. This would simplify a bunch of code in `AutomationView` of the form (`if (clip->type == CLIP_TYPE_AUDIO) { clip->foo(audio->lastSelectedParamID) } else { clip->foo(instrument->lastSelectedParamID) }`